scripts/ensure-frozen-migrations

```diff
@@ -24,39 +21,62 @@ show_diff() {
     diff -u <(remove_comments "$file1") <(remove_comments "$file2")
 }
 
+fail() {
+    echo "❌ Existing migrations changed. Please make sure only comments are changed in existing migration files."
+    exit 1
+}
+
 main() {
     local sql_dir="backend/controller/sql/schema"
     local changed_files
-    local merge_base
+    local all_old_files
+    local new_files=( )
 
-    # Find the merge base commit
-    merge_base=$(git merge-base HEAD origin/main)
+    head_sha=$(git merge-base HEAD origin/main)
+    changed_files=$(git diff --name-only "$head_sha" -- "$sql_dir"/*.sql)
+    all_old_files=$(git diff --name-only origin/main -- "$sql_dir")
 
-    # Get list of changed SQL files compared to the merge base
-    changed_files=$(git diff --name-only "$merge_base" -- "$sql_dir"/*.sql)
+    # Show the diff of the SQL files compared to HEAD to the user
+    git diff --color=always "$head_sha" -- "$sql_dir"/*.sql | cat
 
+    # Track new files added or changed
```

Comment:
What was the actual problem, it's not clear to me?

Reply:
Also shouldn't this use `--exit-code` to just exit if there's a difference?

Reply:
> What was the actual problem, it's not clear to me?

Ah I see - new files were treated as a change, makes sense.

Reply:
(in which case, disregard my `--exit-code` suggestion :) )

Reply:
Is checking for comment changes something we should drop to make this check less complex? It might not be worth it, OTOH the code is already there!

Reply:
I think allowing comment changes is super handy, let's keep it.

Reply:
> What was the actual problem, it's not clear to me?

New files were erroring because they didn't exist in main, and returned an error when doing a diff. I don't know how that happened because I do remember testing it specifically! I should have done the unit test the first time around.